lib/src/Uncertainty/Distribution/Frechet.cxx

```diff
@@ -211,6 +211,21 @@ Point Frechet::computePDFGradient(const Point & point) const
   return pdfGradient;
 }
 
+/* Get the LogPDFGradient of the distribution */
+Point Frechet::computeLogPDFGradient(const Point & point) const
+{
+  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+
+  const Scalar x = point[0] - gamma_;
+  Point logPdfGradient(3);
+  if (x <= 0.0) return logPdfGradient;
+  const Scalar logCdfplus1 = 1.0 - std::pow(x / beta_, -alpha_);
```

Comment:
Here you should use:
```
const Scalar logCdfplus1 = -std::expm1(-alpha_ * std::log(x / beta_));
```
for an improved accuracy when x is close to beta. You can check it using this code:
```
int main(void)
{
  double x = 1.0+1.0/(1<<20); // exact in double precision
  double val1 = 1.0 - std::pow(x, -1.0);
  double val2 = -std::expm1(-1.0 * std::log(x));
  double ref = 9.53673406912415587982570664815e-7; // from Maple
  std::cout << std::setprecision(20) << "x=" << x
            << "\nval1=" << val1 << ", err=" << 1.0-val1/ref
            << "\nval2=" << val2 << ", err=" << 1.0-val2/ref
            << "\nref =" << ref << std::endl;
  return 0;
}
```
which gives:
```
x=1.0000009536743164062
val1=9.5367340691154822707e-07, err=9.0949470177292823792e-13
val2=9.5367340691241558881e-07, err=0
ref =9.5367340691241558881e-07
```

Reply:
Done. Thanks!